src/zkp/piaffg.rs

```diff
@@ -604,12 +608,437 @@ mod tests {
         with_random_paillier_affg_proof(rng, x, y, f)
     }
 
+    #[test]
+    fn piaffg_proof_with_consistent_secret_inputs_out_of_range() -> Result<()> {
+        let mut rng = init_testing();
+        let x_upper_bound = BigNumber::one() << (ELL + EPSILON);
+        let y_upper_bound = BigNumber::one() << (2 ^ (ELL_PRIME + EPSILON));
+        let x = random_plusminus_by_size(&mut rng, ELL);
+        let y = random_plusminus_by_size(&mut rng, 2 ^ ELL_PRIME);
+        loop {
+            let x_too_large = random_plusminus_by_size_with_minimum(
+                &mut rng,
+                ELL + EPSILON + 2,
+                ELL + EPSILON + 1,
+            )?;
+            if x_too_large.gt(&x_upper_bound) {
+                let x_too_small = -x_too_large.clone();
+                let f: TestFn = |bad_proof, input| {
+                    assert!(bad_proof.verify(input, &(), &mut transcript()).is_err());
+                    Ok(())
+                };
+                with_random_paillier_affg_proof(&mut rng, &x_too_large, &y, f)?;
+                with_random_paillier_affg_proof(&mut rng, &x_too_small, &y, f)?;
+                break;
+            }
+        }
+        loop {
```

Comment:
yes precisely. looping so that the too_large number is actually generated sometime. If I omit the loop, then there is no guarantee that the test will actually generate a too large or too small number that we want to test with. So then the sampling function will work but there is also the possibility that most of the times the number generated is less than the too large range. Then there is no point of the test.

Reply:
Oh yeah you are right! Sorry about that, my bad.